# Most Profitable Apps 

   The objective of this project is to recommend a app to build that are profitable in both App Store and Google Play markets.Our main purpose is to build a popular free app to generate revenue from in-app ads. This means our goal is to build an app that attract a great number of users to use.

# Data Source

[Google Play data set](https://www.kaggle.com/lava18/google-play-store-apps) includes data about around 10,000 Android apps from Google Play. 

[Apple Store data set](https://www.kaggle.com/lava18/google-play-store-apps) includes data about around 7000 IOS apps from the APP store.

# Part 1:  Data Cleaning 

## 1. **Exploring the Data**

In [2]:
from csv import reader
# Open the google Play data set #
opened_file = open('C:/Users/jesse/Desktop/New/project/Project 1_Profitable App/googleplaystore.csv',encoding="utf8")
read_file=reader(opened_file)
google = list(read_file)
google_header=google[0]
google = google[1:]
# Open the Apple Store data set #
opened_file = open('C:/Users/jesse/Desktop/New/project/Project 1_Profitable App/AppleStore.csv',encoding="utf8")
read_file=reader(opened_file)
apple = list(read_file)
apple_header=apple[0]
apple = apple[1:]
print(apple_header)
print(google_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Since we want to make it easier to explore, we write a explore_data() function to explore the rows in a more intuitive way.Let's take a look at the Google Play data set and identify the columns that might be useful for this analysis.

In [3]:
def explore_data(dataset, start, end , rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') #make it easier to read
    if rows_and_columns==True:  #can be written as rows_and_columns,that means if true
        print('Number of rows:',len(dataset))
        print('Number of columns:',len(dataset[1]))
print(google_header)
print('\n')
explore_data(google,0,5,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

The Google Play data set has 10841 apps and 13 columns. You can find what each column describes by reading the documentation of the data via this [link](https://www.kaggle.com/lava18/google-play-store-apps). The variables that might be useful for this analysis are 'App','Category','Reviews','Instalss','Type','Price'and 'Genres'.

Then, let's explore the App Store data set.We will also use the explore function on the App Store data set.

In [4]:
print(apple_header)
print('\n')
explore_data(apple,0,5,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0', '2974676', '212', '3.5', '3.5', '95', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0', '1724546', '3842', '4.5', '4', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0', '1126879', '3594', '4', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


As wee can see from the result above, we have 7197 apps in Apple Store and 16 variables. The variables might be useful for this analysis are: 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'prime_genre'. The detailed descriptions for the variables are written in this [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) .

## 2. **Delating Wrong Data**

[One of the discussion]()mentions an error for row 10472. We can explore this in detail.

In [5]:
print(google[10472]) ##row 10472
print('\n')
print(google_header) ##header
print('\n')
print(google[0])   ##print the first line which is the correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up', '']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


As you can see from the result above, for the category column, row 10472 gives a number '1.9', but it was supposed to be a text like the first line. Also, the third position in row 10472 indicates the rating to be 19, but we know the upper limit rating for the Google Play App is 5. Clearly, this row is wrong because of the missing data at the category column in row 10472. Thus, we will delete this row.

In [6]:
print(len(google)) ##we will print again the number of rows in the google data set
del google[10472]  ##delete row 10472 for google data set
print(len(google)) ##check if we successfully delete one row

10841
10840


Now, we have 10840 rows in Google Play data set.

## 3. **Removing Duplicate Entries**

Sometimes, data might have duplicate entries. For instance, the application Instagram has four duplicate entries:

In [7]:
for app_row in google:  ##we want to know how many duplicate entries does instagram have.
    name = app_row[0]
    if name == 'Instagram':
        print(app_row)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As you can see, we have 4 duplicate entries for the Instagram application. Let's figure out how many duplicate apps in the datasets.

In [8]:
duplicate_apps=[]
unique_apps=[]

for each_app in google:
    name=each_app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicate apps:',len(duplicate_apps))
print('\n')
print('Example of duplicate apps:',duplicate_apps[:15])

Number of duplicate apps: 1181


Example of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We want to delete the duplicate rows. As you can see from the instagram example,these rows are not exactly the same.The differences are in the fourth column which is the number of reviews.We want the row with the highest number of reviews, and delete rows with less number of reviews because the higher number of reviews, the more accurate the results.

In order to get the new dataset, we will follow these steps below:
1. Create a dictionary where each key is unique app name, and the value is the highest number of reviews of that app.
2. Use the dictionary to create the new data set, which will have the row with the highest number of reviews of that app.


In [9]:
# building the dictionary
app_dic={} 
for each_row in google:
    name=each_row[0]   
    review=float(each_row[3])
    if name in app_dic and app_dic[name]<review: 
        # meet two requirements:1.the app name is already in the dictionary, and we need to compare if the new # of reviews > the # of
                                # review that is associated with the app already in the dictionary, if yes, replace the # of reviews.
        app_dic[name]=review
    elif name not in app_dic:
        # if the app name is not in the dictionary yet,add this app and the # of reviews to the dictionary.
        app_dic[name]=review
        # else: do not change anything, just move to the next row, until the loop is over.

Let's check if the number of unique apps = the total number of apps(rows) - the number of duplicate apps (1181).        

In [10]:
print(len(app_dic))
number_of_unique_apps=len(google)-1181
print(number_of_unique_apps)
len(app_dic)==number_of_unique_apps

9659
9659


True

In [11]:
# Create the new data set exclude the duplicates
google_new=[]  #create an empty list which will store the unique app row
app_added=[]   #create an empty list which contains apps that are already in the google_new list

for each_row in google:
    name=each_row[0]
    review=float(each_row[3])
    
    if (review==app_dic[name]) and (name not in app_added): 
        # if the review of this iteration = the # of reviews for the app in the dictionary(contains the highest number of reviews)
        # and the name of this app are not in app_added yet(excludes the case which the highest number of reviews for that app appears
        # twice in the google data set)
        google_new.append(each_row)
        app_added.append(name)


Let's explore the new cleaned data set for Google Play Apps data set, and confirm the number of row is 9659.(unique app)

In [12]:
for each_row in google_new:
    if each_row[0]=='Instagram':
        print(each_row)

['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As we can see from the result above and compare the result a few lines up, we only get a single row for the 'Instagram' app with the highest number of review '66577446'.

Let's use again the explore function created above, and confirm the number of rows and columns.

In [13]:
explore_data(google_new,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


As expected, we have 9659 rows and 13 columns.

## 4. Removing Non-English Apps

As mentioned in the data descriptions,there are apps that are not toward English-speaking people. 
For example:

In [14]:
print(apple[6731][1])
print(google_new[7940][0])

【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
لعبة تقدر تربح DZ


Our aim for this analysis is to find the profitable app among English apps, so we will remove the Non-English apps. All the characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, therefore we can build a function to check if an app is an English-app based on the ASCII critiria.

We use the built-in function ord() function to find out the corresponding encoding numbers of each character.

In [15]:
def is_English(app):
    for character in app:
        if 0 <= ord(character)<= 127:
            return False
        else:
            return True
print(is_English('【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜'))
print(is_English('Instagram'))


True
False


Some apps might use emojis or other symbols in their names that fall outside the ASCII range, but these apps are English-apps and we need to include those apps.For Example:

In [16]:
print(is_English('Instachat 😜'))
print(ord('😜'))

False
128540


To minimize the impact of data loss, we'll  only remove the app that has more than 3 non-ASCII characters. Let's write a function to check whether the name of the app contains more than 3 non-ASCII characters.

In [17]:
def is_English_3(app_name):
    non_ASCII_number=0
    for character in app_name:
        if ord(character)>127:
            non_ASCII_number+=1
    if non_ASCII_number>3:
        return False
    else:
        return True
print(is_English_3('Instachat 😜'))

True


Note: Set '3' is not perfect, and we still might filter out some English-apps.

Let's use the is_English_3 function for both data sets.

In [18]:
# create two empty data sets to store the English-apps
google_english=[]  
apple_english=[]

# add the English-apps to the google_english and apple_english lists respectively
for each_row in apple:
    app_name=each_row[1]
    if is_English_3(app_name):
        apple_english.append(each_row)

for each_row in google_new:
    app_name=each_row[0]
    if is_English_3(app_name):
        google_english.append(each_row)

# explore the english data sets and compare it with the original data sets

explore_data(apple,0,3,True) # Orginal Apple Store apps data set
print('\n')
explore_data(apple_english,0,3,True) # English Apps 
print('\n')
explore_data(google,0,3,True)
print('\n')
explore_data(google_english,0,3,True)


['284882215', 'Facebook', '389879808', 'USD', '0', '2974676', '212', '3.5', '3.5', '95', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


['284882215', 'Facebook', '389879808', 'USD', '0', '2974676', '212', '3.5', '3.5', '95', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183
Number of columns: 16


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.

Now we can see that we are left with 6183 apps in the Apple App data set and 9614 apps in the Google Play App data set.

## 5. Removing Non-free Apps

Our aim is to build free apps to generate high revenue with in-app ads. Thus, we need to remove non-free apps in both data sets and then conduct analysis among free apps.

In [19]:
google_free=[]
apple_free=[]

for each_row in google_english:
    fee=each_row[7]
    if fee=='0':
        google_free.append(each_row)

for each_row in apple_english:
    fee=each_row[4]
    if fee=='0':
        apple_free.append(each_row)
print(len(google_free))
print(len(apple_free))
    

8864
3222


We are left with 8864 rows in Google Play apps data set and 3222 rows in Apple apps data set.

# Part 2：Data Analysis

Our aim is to build apps that will be successful on both Google Play and Apple Store.Let's begin the analysis by getting the most common genres for each market.

## 1. Common Apps by Genre ( Frequency Tables )

We will build two functions we can use to analyze the frequency tables:
  1. A function to generate frequency tables that show percentages
  2. A function to discplay the frequency tables by percentages in a descending order

In [20]:
# Create a function to generate frequency tables that show percentages

def freq_t(data_set,index):
    freq_table={}
    total=len(data_set)
    for each_row in data_set:
        genre=each_row[index]
        if genre in freq_table:
            freq_table[genre]+=1
        else:
            freq_table[genre]=1

    per_t={}
    for element in freq_table:
        per_t[element]=(freq_table[element]/len(data_set))*100
    
    return per_t

# Show the frequency tables for Genre associated with the percentages
result_apple=freq_t(apple_free,-5)
print(result_apple)
print('\n')
result_google=freq_t(google_free,-4)
print(result_google)

{'Social Networking': 3.2898820608317814, 'Photo & Video': 4.9658597144630665, 'Games': 58.16263190564867, 'Music': 2.0484171322160147, 'Reference': 0.5586592178770949, 'Health & Fitness': 2.0173805090006205, 'Weather': 0.8690254500310366, 'Utilities': 2.5139664804469275, 'Travel': 1.2414649286157666, 'Shopping': 2.60707635009311, 'News': 1.3345747982619491, 'Navigation': 0.186219739292365, 'Lifestyle': 1.5828677839851024, 'Entertainment': 7.883302296710118, 'Food & Drink': 0.8069522036002483, 'Sports': 2.1415270018621975, 'Book': 0.4345127250155183, 'Finance': 1.1173184357541899, 'Education': 3.662321539416512, 'Productivity': 1.7380509000620732, 'Business': 0.5276225946617008, 'Catalogs': 0.12414649286157665, 'Medical': 0.186219739292365}


{'Art & Design': 0.5979241877256317, 'Art & Design;Creativity': 0.06768953068592057, 'Auto & Vehicles': 0.9250902527075812, 'Beauty': 0.5979241877256317, 'Books & Reference': 2.1435018050541514, 'Business': 4.591606498194946, 'Comics': 0.609205776

In this form, it is difficult to find the most common apps by Genre.Therefore, we will write a function to display the percentages in descending order.

In [21]:
def display_table(data_set,index):
    table=freq_t(data_set,index)
    table_display=[]
    for key in table:
        each_row= table[key],key
        table_display.append(each_row)
    
    table_sorted=sorted(table_display,reverse=True)
    
    for element in table_sorted:
        print(element[1],":",element[0])

In [22]:
print('Frequency Table by Prime Genres in Pencentages for Apple Apps:')
display_table(apple_free,-5)
print('\n')
print('Frequency Table by Genres in Pencentages for Google Play Apps:')
display_table(google_free,-4)
print('\n')
print('Frequency Table by Category in Pencentages for Google Apps:')
display_table(google_free,1)


Frequency Table by Prime Genres in Pencentages for Apple Apps:
Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Frequency Table by Genres in Pencentages for Google Play Apps:
Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.89

As we can see the two data sets has different column name for Genre.In Google Play data set, the Category column is similar to the Prime Genres column in Apple Apps data set. Thus, we will use the Category column in Google Play data set.

The two markets have different answers for the most common genre:

1. In Apple Apps data set(English),58.16% are games.Entertainment apps has the percentage of 8%; photo and video apps are close to 5%. Only 3.66% of the apps are education, and 3.29% of the apps are social networking apps. The apps in Apple Store is dominated by apps that are designed for fun(ex:games,entertainment,photo and video,etc.) The apps that focus on the education are less common. 

2. In Google Play data set(English),the result shows many apps are desiged for family,tools,business,lifestyle,productivity,etc. This result is quite different from what we got from the Apple apps data set.The genres are more balanced than Apple Apple.


Keep in mind, the most common genre does not mean it has more number of users than other genres.Next, let's look at the most popular apps by genres.

## 2. Most Popular Apps by Genre

* App Store

For App Store data set, we don't have the Installs column,so we will use the total number of user ratings as a proxy which is the rating_count_tot column in the App Store data set.

We calculate the average number of user ratings per app genre on the App Store:

In [23]:
genres_app=freq_t(apple_free,-5)  # make a frequency table for apple sotre: the first column is genre, and second column is percentages.

for genre in genres_app:   # for each genre in this frequency table
    total=0   # store the total number of ratings in a genre
    len_genre=0   # store the number of apps in this type of genre
    for app in apple_free:  # for each app in apple_free data set
        app_genre=app[-5]  
        if app_genre == genre: # if genre = genre in frequency table
            number_ratings=float(app[5])  # number_ratings is the 5th column of the data set
            total+=number_ratings  # add the number of rating for each app in the total
            len_genre += 1   # count how many apps are in this genre
    avg_number_ratings=total/len_genre   # calculate the average number of ratings for this genre
    print(genre,":",avg_number_ratings) # print the result
    

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


As we can see that the Navigation apps have the highest number of ratings,and we know that this is heavily affected by Waze and Google Maps.

In [24]:
# Let's take a look at the number of ratings for navigation APPs:
for each_row in apple_free:
    if each_row[-5]=='Navigation':
        print(each_row[1],":",each_row[5])  

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [25]:
# The number of ratings for Social Networking Apps:
for each_row in apple_free:
    if each_row[-5]=='Social Networking':
        print(each_row[1],":",each_row[5])


Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [27]:
# The number of ratings for Music Apps:
for each_row in apple_free:
    if each_row[-5]=='Music':
        print(each_row[1],":",each_row[5])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

As you can see among music and social networking apps,the averages of the number of ratings are highly influenced by the popular social networking apps and the music apps, for example, Facebook, Pinterest, Skype, Pandora, Spotify, etc. The average number of ratings seem to be infuluenced by most popular app in that genre.

One thing to keep in mind, the App store is dominated by the for-fun apps, and the practical app might have higher chances to stand out.

* Google Play Store

Let's take a glance at the Google play data set below. We have the number of installs column, but the install numbers are not very precise,mostly are open-ended (for example:100+,1,000+,5,000+, etc). Our aim is to get some senses on which app genres are most popular, so we will use the data as they are but remove the ',' and '+' for each data.

In [28]:
display_table(google_free,5) # the installs column

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


Let's write a function to remove ','and '+' and create a frequency table to display the average number of installs in each category.

In [32]:
categories_google = freq_t(google_free,1) # the category column

for category in categories_google:
    len_category=0
    total_installs=0
    for each_row in google_free:
        app_category=each_row[1]
        n_installs=each_row[5]
        if category == app_category:
            len_category+=1
            n_installs=n_installs.replace(",","")
            n_installs=n_installs.replace('+',"")
            total_installs+=float(n_installs)
    average_n_installs = total_installs/len_category
    print(category,":",average_n_installs)
     

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Communication apps have the most installs on average.Let's take a look at the number of installs for the top communication apps.

In [34]:
for each_row in google_free:
    if each_row[1] == 'COMMUNICATION' and (each_row[5] == '1,000,000,000+'
                                          or each_row[5] == '500,000,000+'
                                          or each_row[5] == '100,000,000+'):
        print(each_row[0],":",each_row[5])
    

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

From the above results, we notice that the average number of installs of communication apps is highly influenced by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts). Let's remove all the communication apps that have over 100 million installs, and have a look at the average number again.

In [36]:
under_100m = []

for each_row in google_free:
    n_installs=each_row[5]
    n_installs=n_installs.replace("+","")
    n_installs=n_installs.replace(",","")
    if (each_row[1] == "COMMUNICATION") and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))

avg_under_100m=sum(under_100m)/len(under_100m)
print(avg_under_100m)

3603485.3884615386


After removing all the communication apps that have over 100 million installs, the average number of installs of the communication apps has decreased a lot. The same patterns for video players category since the market is dominated by apps like Youtube, Google Play Movies & TV etc. 

We will not recommend to build new app in these categories since it is hard to compete against the top apps within the category. Let's look at the books and reference genre and see if this category is a good option.

In [38]:
for each_row in google_free:
    if each_row[1] == 'BOOKS_AND_REFERENCE':
        print(each_row[0], ':',each_row[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

Let's see if this market has been dominated by popular apps.

In [42]:
for each_row in google_free:
    if (each_row[1] == 'BOOKS_AND_REFERENCE') and (each_row[5]=='1,000,000,000+'
                                                  or each_row[5]=='500,000,000+'
                                                  or each_row[5]=='100,000,000+'):
        print(each_row[0],':',each_row[5])


Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


We see some potential in this market since the number of apps that have extremely large number of installs are less than categories we discussed above.

Let's analyze the apps in this category and with less installs (between 1,000,000 and 100,000,000 downloads).

In [43]:
for each_row in google_free:
    if each_row[1] == "BOOKS_AND_REFERENCE" and (each_row[5]=='1,000,000+'
                                                or each_row[5]== '5,000,000+'
                                                or each_row[5]=='10,000,000+'
                                                or each_row[5]== '50,000,000+'):
        print(each_row[0],":",each_row[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

These apps are created for processing and reading ebooks, libraries and dictionaries, so it might be a good idea to avoid building similar app.

It might be profitable to build an app around a popular and recent book in both the Google Play and the App Store, and we need to think of a more creative way to turn a book into an app.


# Part 3：Conclusion

Our aim is to build a profitable App in both App Store and Google Play markets. After analyzing the popularity of genres, we conclude that it might be profitable to turn a popular and recent book into an app with more creative and fun features. 